File: app/src/main/java/mydebts/android/app/data/PersonsSource.java
package mydebts.android.app.data;

import android.support.annotation.NonNull;

import java.util.List;

import io.reactivex.Observable;
import io.reactivex.Single;
import mydebts.android.app.data.model.Person;

public interface PersonsSource {

    Single<List<Person>> getAll();

    Single<Person> get(@NonNull Long id);

    Observable<Person> inserted();

    Observable<Person> updated();

    Observable<Person> deleted();
}

File: app/src/main/java/mydebts/android/app/data/PersonsSource.kt
package mydebts.android.app.data

import io.reactivex.Single
import mydebts.android.app.data.model.Person

interface PersonsSource {

    fun getAll(): Single<List<Person>>

    fun get(id: Long): Single<Person>

    fun insert(person: Person): Single<Person>

    fun update(person: Person): Single<Person>

    fun delete(person: Person): Single<Person>
}
